src/scipp/logging.py

```diff
@@ -0,0 +1,183 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Jan-Lukas Wynen
+"""
+Utilities for managing scipp's logger and log widget.
+"""
+
+import html
+import logging
+import time
+from typing import Optional
+
+from .utils import running_in_jupyter
+from ._styling import inject_style
+
+
+def get_logger() -> logging.Logger:
+    """
+    Return the global logger used by scipp.
+    """
+    logger = logging.getLogger('scipp')
+    if not get_logger.is_set_up:
+        _setup_logger(logger)
+        get_logger.is_set_up = True
+    return logger
+
+
+get_logger.is_set_up = False
+
+
+def _setup_logger(logger: logging.Logger) -> None:
+    logger.setLevel(logging.INFO)
+    logger.addHandler(make_stream_handler())
+    if running_in_jupyter():
+        logger.addHandler(make_widget_handler())
```

Comment:
I suppose sooner or later we will want to add ability to log to a file. Does this mechanism allow for adding this easily? I am not suggesting to do so now.

Reply:
Yes, it is easy
```.py
sc.get_logger().addHandler(logging.FIleHandler('<filename>'))
```
I don't think that scipp should write to a file by default. This would only result in either cluttering the working directory or leaving behind files where no one knows what program they belong to.

Reply:
Definitely, this should only even be done on demand.

Reply:
After discussing this, we decided that since we refer to the documentation of the standard library logger, we don't need to document this further.